ragna/deploy/_api/core.py

```diff
@@ -145,6 +145,19 @@ async def get_components(_: UserDependency) -> schemas.Components:
             ],
         )
 
+    @app.get("/corpuses")
+    async def get_corpuses(_: UserDependency) -> dict[str, list[str]]:
```

Comment:
Following the discussion in https://github.com/Quansight/ragna/issues/493#issuecomment-2301776184 should we introduce a `source_storage` query parameter?

```suggestion
    async def get_corpuses(_: UserDependency, source_storage: Optional[str] = None) -> dict[str, list[str]]:
```

And with that we can do

```python
if source_storage is not None:
    source_storages = [components_map[source_storage]]
else:
    ...
```

Meaning, if the user does `GET /corpuses` they get everything, but they can do `GET /corpuses?source_storage=Chroma` to pre-filter it. Still, the output format should be the same in both cases.

Reply:
I like that. I'll push that up.